src/sys/windows/afd.rs

```diff
@@ -1,17 +1,28 @@
-use ntapi::ntioapi::{IO_STATUS_BLOCK_u, IO_STATUS_BLOCK};
-use ntapi::ntioapi::{NtCancelIoFileEx, NtDeviceIoControlFile};
-use ntapi::ntrtl::RtlNtStatusToDosError;
+use std::ffi::c_void;
 use std::fmt;
 use std::fs::File;
 use std::io;
 use std::mem::size_of;
 use std::os::windows::io::AsRawHandle;
-use std::ptr::null_mut;
-use winapi::shared::ntdef::{HANDLE, LARGE_INTEGER, NTSTATUS, PVOID, ULONG};
-use winapi::shared::ntstatus::{STATUS_NOT_FOUND, STATUS_PENDING, STATUS_SUCCESS};
-
-const IOCTL_AFD_POLL: ULONG = 0x00012024;
 
+use windows_sys::Win32::{
+    Foundation::{
+        RtlNtStatusToDosError, HANDLE, NTSTATUS, STATUS_NOT_FOUND, STATUS_PENDING, STATUS_SUCCESS,
+    },
+    System::WindowsProgramming::{NtDeviceIoControlFile, IO_STATUS_BLOCK, IO_STATUS_BLOCK_0},
+};
+
+const IOCTL_AFD_POLL: u32 = 0x00012024;
+
+// <https://processhacker.sourceforge.io/doc/ntioapi_8h.html#a0d4d550cad4d62d75b76961e25f6550c>
+#[link(name = "ntdll")]
+extern "system" {
```

Comment:
Should this be a part of `windows-sys`?

Reply:
This appears to be an undocumented API that does not appear in the Windows SDK. The `windows-sys` crate is generated from metadata for the Windows SDK. You can [request it be added here](https://github.com/microsoft/win32metadata) but given that it is undocumented, it may not be accepted.